kroxylicious-test-tools/src/main/java/io/kroxylicious/test/tester/KroxyliciousClients.java

```diff
@@ -64,14 +77,27 @@ public Consumer<String, String> consumer() {
 
     public <U, V> Consumer<U, V> consumer(Serde<U> keySerde, Serde<V> valueSerde, Map<String, Object> additionalConfig) {
         Map<String, Object> config = createConfigMap(bootstrapServers, additionalConfig);
-        return CloseableConsumer.wrap(new KafkaConsumer<>(config, keySerde.deserializer(), valueSerde.deserializer()));
+        Consumer<U, V> consumer = CloseableConsumer.wrap(new KafkaConsumer<>(config, keySerde.deserializer(), valueSerde.deserializer()));
+        consumers.add(consumer);
+        return consumer;
     }
 
     public KafkaClient simpleTestClient() {
         String[] hostPort = bootstrapServers.split(":");
         return new KafkaClient(hostPort[0], Integer.parseInt(hostPort[1]));
     }
 
+    public void close() {
+        try {
```

Comment:
It might be good to attempt to shutdown all the clients even if some throw an Exception on close, so you'd try-catch in the for-each but still have some mechanism to throw an overall exception if something failed.

Reply:
Yeah, definitely want to keep closing even if one client throws. 

On that would users care which client failed to close? Do we need to track failures or do we just need a throw if any fail?

Reply:
In my mind it makes sense to just re-throw whatever was thrown first (i.e. whatever ended up in the list first) once we've closed everything, because chances are whatever went wrong is going to be there. There's also not really a neat way I could find to translate multiple arbitrary exceptions into one coherent `RuntimeException`, so I figured it would make the most sense to re-throw whatever went wrong first.